exporters/otlp/otlpmetric/internal/otest/client.go

```diff
@@ -272,6 +272,33 @@ func RunClientTests(f ClientFactory) func(*testing.T) {
 			want := fmt.Sprintf("%s (%d metric data points rejected)", msg, n)
 			assert.ErrorContains(t, errs[0], want)
 		})
+
+		t.Run("Other HTTP success", func(t *testing.T) {
+			for i := 201; i <= 299; i++ {
+				t.Run(fmt.Sprintf("status_%d", i), func(t *testing.T) {
+					rCh := make(chan ExportResult, 1)
+					rCh <- ExportResult{
+						ResponseStatus: i,
+					}
+
+					ctx := context.Background()
+					client, _ := f(rCh)
+
+					defer func(orig otel.ErrorHandler) {
+						otel.SetErrorHandler(orig)
+					}(otel.GetErrorHandler())
+
+					errs := []error{}
+					eh := otel.ErrorHandlerFunc(func(e error) { errs = append(errs, e) })
+					otel.SetErrorHandler(eh)
+
+					require.NoError(t, client.UploadMetrics(ctx, nil))
+					require.NoError(t, client.Shutdown(ctx))
+
+					require.Equal(t, 0, len(errs))
```

Comment:
I think we can change `require` to `assert` as any of the assertions does not need to interrupt the test.